pandas/io/formats/format.py

```diff
@@ -1324,6 +1326,12 @@ def _get_format_datetime64(is_dates_only, nat_rep='NaT', date_format=None):
 
 def _get_format_datetime64_from_values(values, date_format):
     """ given values and a date_format, return a string format """
+
+    if isinstance(values, np.ndarray) and values.ndim > 1:
+        # We don't actaully care about the order of values, and DatetimeIndex
+        #  only accepts 1D values
+        values = values.ravel()
```

Comment:
I would even move it down one step further to `to_native_types` (there you don't need the if check, other part of the code there is already doing `ravel()` as well), but no strong feelings if you want to keep it here

Reply:
@jbrockmendel